tailfs/connlistener.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package tailfs
+
+import (
+	"net"
+	"sync"
+	"syscall"
+)
+
+type connListener struct {
+	ch       chan net.Conn
+	closedCh chan interface{}
+	closeMu  sync.Mutex
+}
+
+// newConnListener creates a net.Listener to which one can hand connections
+// directly.
+func newConnListener() *connListener {
+	return &connListener{
+		ch:       make(chan net.Conn),
+		closedCh: make(chan interface{}),
+	}
+}
+
+func (l *connListener) Accept() (net.Conn, error) {
+	select {
+	case <-l.closedCh:
+		// TODO(oxtoacart): make this error match what a regular net.Listener does
+		return nil, syscall.EINVAL
+	case conn := <-l.ch:
+		return conn, nil
+	}
+}
+
+func (l *connListener) Addr() net.Addr {
```

Comment:
I think now that it's unexported, we no longer need this to implement `net.Listener` at all and can drop this method entirely.